src/config_creature.c

```diff
@@ -821,29 +827,49 @@ TbBool parse_creaturetype_experience_blocks(char *buf, long len, const char *con
 
 TbBool parse_creaturetype_instance_blocks(char *buf, long len, const char *config_textname, unsigned short flags)
 {
+    struct InstanceInfo* inst_inf;
     int i;
     // Block name and parameter word store variables
-    int arr_size;
     // Initialize the array
-    if ((flags & CnfLd_AcceptPartial) == 0)
+    int arr_size = sizeof(gameadd.crtr_conf.instances) / sizeof(gameadd.crtr_conf.instances[0]);
+    for (i = 0; i < arr_size; i++)
     {
-        arr_size = sizeof(gameadd.crtr_conf.instances)/sizeof(gameadd.crtr_conf.instances[0]);
-        for (i=0; i < arr_size; i++)
+        if (((flags & CnfLd_AcceptPartial) == 0) || (strlen(gameadd.crtr_conf.instances[i].name) <= 0))
         {
             LbMemorySet(gameadd.crtr_conf.instances[i].name, 0, COMMAND_WORD_LEN);
             if (i < gameadd.crtr_conf.instances_count)
             {
                 instance_desc[i].name = gameadd.crtr_conf.instances[i].name;
                 instance_desc[i].num = i;
-            } else
+                magic_conf.instance_info[i].instant = 0;
+                magic_conf.instance_info[i].time = 0;
+                magic_conf.instance_info[i].fp_time = 0;
+                magic_conf.instance_info[i].action_time = 0;
+                magic_conf.instance_info[i].fp_action_time = 0;
+                magic_conf.instance_info[i].reset_time = 0;
+                magic_conf.instance_info[i].fp_reset_time = 0;
+                magic_conf.instance_info[i].graphics_idx = 0;
+                magic_conf.instance_info[i].flags = 0;
+                magic_conf.instance_info[i].force_visibility = 0;
+                magic_conf.instance_info[i].primary_target = 0;
+                magic_conf.instance_info[i].func_cb = 0;
+                magic_conf.instance_info[i].func_params[0] = 0;
+                magic_conf.instance_info[i].func_params[1] = 0;
+                magic_conf.instance_info[i].symbol_spridx = 0;
+                magic_conf.instance_info[i].tooltip_stridx = 0;
+                magic_conf.instance_info[i].range_min = -1;
+                magic_conf.instance_info[i].range_max = -1;
```

Comment:
would it be a better default to set range_min and range_max to 0 and LONG_MAX, in case you didn't add the info in the config?

Reply:
The get_best_combat_weapon_instance_to_use and get_best_combat_weapon_instance_to_use_versus_trap fields check all instances with a range, this would include weapons with ranges at min 0. The ones with -1 are excluded there now.